game/modes/new_moon.py

```diff
@@ -45,3 +52,56 @@ def get_current_event_description(self):
 
     def do_coin_toss(self):
         return random.randint(0, 1)
+
+    async def send_result_text(self, interface, channel_name, **kwargs):
+        if self.has_special_event():
+            await interface.send_action_text_to_channel(
+                f"new_moon_{self.current_event}_result_text", channel_name, **kwargs
+            )
+
+    async def do_action(self, interface, **kwargs):
+        if self.current_event == NewMoonMode.HEADS_OR_TAILS:
+            await self.do_heads_or_tails_action(interface, kwargs.get("coin_toss_value"))
+
+        elif self.current_event == NewMoonMode.TWIN_FLAME:
+            await self.do_twin_flame_action(interface, kwargs.get("cupid_target"))
+
+        elif self.current_event == NewMoonMode.FULL_MOON_VEGETARIAN:
+            await self.do_full_moon_vegetarian_action(interface)
+
+        elif self.current_event == NewMoonMode.SOMNAMBULISM:
+            await self.do_somnambulism_action(interface, kwargs.get("target"))
+
+        elif self.current_event == NewMoonMode.PUNISHMENT:
+            await self.do_punishment_action(interface, kwargs.get("author"), kwargs.get("target"))
```

Comment:
We should change the coding style from `if elif` to early return
```suggestion
    async def do_action(self, interface, **kwargs):
        if self.current_event == NewMoonMode.HEADS_OR_TAILS:
            return await self.do_heads_or_tails_action(interface, kwargs.get("coin_toss_value"))

        if self.current_event == NewMoonMode.TWIN_FLAME:
            return await self.do_twin_flame_action(interface, kwargs.get("cupid_target"))

        if self.current_event == NewMoonMode.FULL_MOON_VEGETARIAN:
            return await self.do_full_moon_vegetarian_action(interface)

        if self.current_event == NewMoonMode.SOMNAMBULISM:
            return await self.do_somnambulism_action(interface, kwargs.get("target"))

        if self.current_event == NewMoonMode.PUNISHMENT:
            return await self.do_punishment_action(interface, kwargs.get("author"), kwargs.get("target"))
        
        # Log unknown error flow here for future debug
        print("Error: Unknown NewMoonMode event")
        return
```